File: Android/dokit-no-op/src/main/java/com/didichuxing/doraemonkit/kit/network/bean/Request.java
package com.didichuxing.doraemonkit.kit.network.bean;

import android.text.TextUtils;

import java.io.Serializable;

/**
 * 请求bean
 */
public class Request implements Serializable {

    public String url;

    public String method;

    public String headers;

    public String postData;

    public String encode;

    @Override
    public String toString() {
        return "";
    }

    public boolean filter(String text) {
        return false;
    }
}

File: Android/kotlin/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/network/okhttp/bean/Request.kt
package com.didichuxing.doraemonkit.kit.network.okhttp.bean

import android.text.TextUtils
import java.io.Serializable

/**
 * 请求bean
 */
class Request : Serializable {
    var url: String? = null
    var method: String? = null
    var headers: String? = null
    var postData: String? = null
    var encode: String? = null
    override fun toString(): String {
        return String.format("[%s %s %s %s]", url, method, headers.toString(), postData)
    }

    fun filter(text: String?): Boolean {
        return !TextUtils.isEmpty(url) && url!!.contains(text!!)
    }
}